### DISCUSSION TITLE
rauc-hawkbit-updater fails to install software bundle to raspberry pi

### DISCUSSION THREAD

**Discussion by erhan-namli** [rauc/rauc#1349] [Q&A]
Hello, I am running the rauc-hawkbit-updater program on my Raspberry Pi and hawkbit is running on my Ubuntu computer on the same local network. When I try to send an application by Hawkbit to my device I get an error like this:,

```
$ ./rauc-hawkbit-updater -c config.conf
MESSAGE: Checking for new software...
MESSAGE: New software ready for download (Name: ttcontrol, Version: 1.0, Size: 649640 bytes, URL: http://192.168.0.130:8080/DEFAULT/controller/v1/rpi-test-3/softwaremodules/1/artifacts/ttControl)
MESSAGE: Start downloading: http://192.168.0.130:8080/DEFAULT/controller/v1/rpi-test-3/softwaremodules/1/artifacts/ttControl
MESSAGE: Download complete. 6.52 MB/s
MESSAGE: File checksum OK.
MESSAGE: Checking for new software...
WARNING: Failed to create RAUC DBUS proxy: Error calling StartServiceByName for de.pengutronix.rauc: Timeout was reached
WARNING: Failed to install software bundle.
```

by the way, I made the config settings of the rauc hawkbit-updater correctly.

Do I need to install the rauc repo on my raspberry pi?  I tried to install it, but the rauc service gave an error while installing it. For example, system.conf is not automatically created under etc/rauc and I could not manage to create it myself...

Where could I be making a mistake?

**Comment by ejoerns** [comment]
@erhan-namli Hi. Yes, rauc-hawkbit-updater is just the service that interfaces with hawkBit and RAUC. So you need to have RAUC running on the target.

For information on how to integrate RAUC into your image, see doc chapter https://rauc.readthedocs.io/en/latest/integration.html

How did you try to install it? Do you use some sort of binary Linux distribution like debian or Raspian?

The `system.conf` very much depends on your partitioning. It won't be auto-created. There are some examples shown here https://rauc.readthedocs.io/en/latest/examples.html#example-slot-configurations that you might need to adapt to your use case.

Note that for raspberrypi there also is a Yocto/OE example layer that might help or give inspiration: https://github.com/rauc/meta-rauc-community/tree/master/meta-rauc-raspberrypi

**Reply by erhan-namli** [reply]
Thank you for your answer, I will try something more by following the information in the link you sent.

For your questions, I am using the Raspbian operating system, and I followed the building from source steps in the rauc repository to run rauc on the system.
